### PR TITLE
[PATCH API-NEXT v2] api: crypto: clarify for odp_crypto_session_create

### PR DESCRIPTION
**Pull request by lumag**
Signed-off-by: Dmitry Eremin-Solenikov <dmitry.ereminsolenikov@linaro.org>

### CODE REVIEW

File: include/odp/api/spec/crypto.h [hunk 1]
```diff
@@ -647,7 +647,8 @@ int odp_crypto_auth_capability(odp_auth_alg_t auth,
  *
  * Create a crypto session according to the session parameters. Use
  * odp_crypto_session_param_init() to initialize parameters into their
- * default values.
+ * default values. If call ends up with an error, 'session' argument won't be
+ * updated and no new session will be created.
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/crypto.h]: I'm not sure how this is a clarification. The documentation for the session parameter below clearly states it is set to the created session or ODP_CRYPTO_SESSION_INVALID, so in either case it's an output parameter and is set by this API.

File: include/odp/api/spec/crypto.h [hunk 2]
```diff
@@ -647,7 +647,8 @@ int odp_crypto_auth_capability(odp_auth_alg_t auth,
...
  *
  * @param param             Session parameters
  * @param session           Created session else ODP_CRYPTO_SESSION_INVALID
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/crypto.h]: The documentation change needed here is that both session and status should be marked @param[out]

**Comment by Bill-Fischofer-Linaro** [comment]
Looks like one of the Travis checks has failed, but I'm not sure why. Needs to be investigated. Also @psavol should review API spec wording change.

**Comment by Bill-Fischofer-Linaro** [comment]
And once again, asking Travis to re-run that "failing" test causes it to pass. Looks like we definitely have some intermittent failure issues.

**Comment by muvarov** [comment]
@psavol  is that ok?

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**